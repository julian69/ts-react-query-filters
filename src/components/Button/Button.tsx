import { FC, SyntheticEvent } from 'react';

interface ButtonProps {
  onClick: (e: SyntheticEvent) => void;
  value: string;
  color?: string;
  isDisabled?: boolean;
  type?: 'submit' | 'button';
}

const Button: FC<ButtonProps> = ({ onClick, value, type, isDisabled, color }) => {
  return (
    <button
      disabled={isDisabled}
      type={type || 'submit'}
      onClick={e => onClick(e)}
      className={`rounded-lg ${color || 'bg-blue-600 text-white'} text-sm px-8 py-2 font-medium  outline-none hover:opacity-80 focus:ring" ${isDisabled ? 'opacity-50' : ''}`}>
      {value}
    </button>
  );
};

export default Button;

