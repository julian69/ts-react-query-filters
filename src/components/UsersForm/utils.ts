import { IFilters } from "@/types/users";

export const deserializeUrl = (params: URLSearchParams): IFilters => ({
  name: params.get('name') || '',
  email: params.get('email') || '',
  company: params.get('company') || '',
  zipcode: params.get('zipcode') || '',
});


