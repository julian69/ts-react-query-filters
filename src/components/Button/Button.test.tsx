import '@testing-library/jest-dom';
import { render } from '@/test/utils';
import { describe, expect, it, vi } from 'vitest';

import Button from './Button.tsx'

describe('<Button />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button value="test" onClick={vi.fn}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});