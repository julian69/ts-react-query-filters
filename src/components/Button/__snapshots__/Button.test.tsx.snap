// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render correctly 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg bg-blue-600 text-white text-sm px-8 py-2 font-medium  outline-none hover:opacity-80 focus:ring" "
    type="submit"
  >
    test
  </button>
</DocumentFragment>
`;
